bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableDropTarget.java

```diff
@@ -201,30 +204,31 @@ private boolean dropNodes(final int action, final Transferable transferable, fin
 
     /** Return true if its okay to drop the item(s) here */
     private boolean isAcceptableDropLocation(final DropTargetDragEvent dtde) {
-        if (!_tree.isEnabled()) {
+        if (!table.isEnabled()) {
             return false;
         }
 
         // Get the node under the mouse.
-        final TelescopePosTableWidget.TableData.Row parent = _tree.getNode(dtde.getLocation());
+        final TelescopePosTableWidget.TableData.Row parent = table.getNode(dtde.getLocation());
         if (parent == null) {
             return false;
         }
 
         // Reject if it would move a node that should not be deleted.
         final TelescopePosTableDragDropObject ddo = TelescopePosTableDragSource._dragObject;
-        return (ddo != null) && (ddo.getOwner() == _tree) && _tree.isOkayToMove(ddo.getNode(), parent);
+        return (ddo != null) && (ddo.getOwner() == table) && table.isOkayToMove(ddo.getNode(), parent);
     }
 
-    /** Save the current tree selection: we only allow one node to be selected. */
-    private void saveTreeSelection() {
-        _selection = IntStream.of(_tree.getSelectedRows()).mapToObj(_tree::getPathForRow).findFirst();
-        treeIgnoreSelectionAction(_tree::clearSelection);
+    /** Save the current table selection: we only allow one row to be selected. */
+    private void saveTableSelection() {
+        final int idx = table.getSelectedRow();
+        _selection = idx == -1 ? None.instance() : new Some<>(idx);
+        treeIgnoreSelectionAction(table::clearSelection);
     }
 
-    /** Restore the tree selection: we only allow one node to be selected. */
-    private void restoreTreeSelection() {
-        treeIgnoreSelectionAction(() -> _selection.map(_tree::getRowForPath).ifPresent(_tree::selectRowAt));
+    /** Restore the table selection: we only allow one row to be selected. */
+    private void restoreTableSelection() {
+        treeIgnoreSelectionAction(() -> _selection.foreach(table::selectRowAt));
```

Comment:
It wasn't as bad as you'd think.

Guide stars can only be dragged from one group to another (otherwise the drag op fails), and user targets cannot be dragged into anything at all, which is what I think is the expected behaviour.